### ISSUE TITLE
package naming causes test run not to load applicationcontext 

### ISSUE THREAD

**Issue by SanjayGautam** [spring-projects/spring-boot#13574]
Dear Contributors, 

I have observed a weird behavior in the spring boot test context resolution. 

how is my project laid out? 
**WHAT DOES NOT WORK**
1. src/main/java

```
    com.caiso.omsint
         Class with @SpringBootApplication

    com.caiso.omsint.config
         Java Confgiuration here  OmsoatiApplicationConfig
         @ComponentScan(basePackages = { "com.cais**o.omsoati.*"**, "com.caiso.omsint.*", 

"com.caiso.soa.saas.*" }, excludeFilters = {
        @ComponentScan.Filter(type = FilterType.REGEX, pattern = { "com.caiso.omsint.config.*" }),
        @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, value = Configuration.class) })

    com.caiso.omsint.dao

    com.caiso.omsint.model

    com.caiso.**omsoati**.b2b.endpoint
                     Class which is endpoint   @Endpoint

     com.caiso**.omsoati**.b2b.endpoint
```

2 .  src/main/test

```
      com.caiso.omsint

 com.caiso.**.omsoati**.a2a.endpoint
         Trying to Test CLass with Endpoint. 
          Using  @RunWith(SpringJUnit4ClassRunner.class)
                    @SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
```

**when I run the unit test case it is not able to load the application context.** 

WHAT WORKS 
1. src/main/java
```
    com.caiso.omsint
         Class with @SpringBootApplication

    com.caiso.omsint.config
         Java Confgiuration here  OmsoatiApplicationConfig
         @ComponentScan(basePackages = { "com.cais**o.omsoati.*"**, "com.caiso.omsint.*", 

"com.caiso.soa.saas.*" }, excludeFilters = {
        @ComponentScan.Filter(type = FilterType.REGEX, pattern = { "com.caiso.omsint.config.*" }),
        @ComponentScan.Filter(type = FilterType.ASSIGNABLE_TYPE, value = Configuration.class) })

    com.caiso.omsint.dao

    com.caiso.omsint.model

    com.caiso.**omsint**.b2b.endpoint
                     Class which is endpoint   @Endpoint

     com.caiso**.omsint**.b2b.endpoint
```

2 .  src/main/test

```      
com.caiso.omsint
      
com.caiso.**.omsint**.a2a.endpoint
         Trying to Test CLass with Endpoint. 
          Using  @RunWith(SpringJUnit4ClassRunner.class)
                    @SpringBootTest(webEnvironment = WebEnvironment.DEFINED_PORT)
```

**when I run the unit test case it loads the application context.** 

My question is that inspite of providing the   @ComponentScan(basePackages permission why it is still not able to load the application context , Some wierd issue which I keep facing time and time agains.  

any guidance would be highly appreciated. I believe it is an issue, it should work when we provide the component scan permission.

**Comment by wilkinsona** [comment]
Thanks for the report. I’m finding the description rather hard to follow. Can you please create a minimal sample project (something that we can unzip or git clone and run) that recreates the problem?

**Comment by philwebb** [comment]
This could be related to missing `@ComponentScan` filters. The `@SpringBootApplication` uses the following:

```java
@Filter(type = FilterType.CUSTOM, classes = TypeExcludeFilter.class),
@Filter(type = FilterType.CUSTOM, classes = AutoConfigurationExcludeFilter.class) })
```

Without a sample, it's quite hard to understand the issue I'm afraid.

**Comment by spring-issuemaster** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by spring-issuemaster** [comment]
Closing due to lack of requested feedback. If you would like us to look at this issue, please provide the requested information and we will re-open the issue.
